### PR TITLE
allow user to choose wider editor

### PR DESCRIPTION
**Pull request by smallst**
for 800-1000px
orig: 
image
new: 
image

for 1020 px 
orig: 
image
new:
image

1470px orig:
image
new:
image

the user can change it in options: 
image

### CODE REVIEW

File: app/views/play/level/PlayLevelView.coffee [hunk 1]
```diff
@@ -835,9 +835,19 @@ module.exports = class PlayLevelView extends RootView
       when @level?.isType('web-dev') then windowHeight - controlBarHeight
       when tomeLocation is 'bottom' then Math.min(windowHeight - minTomeHeight - controlBarHeight, windowWidth / canvasAspectRatio)
       else Math.min(windowHeight - (if controlBarLocation is 'left' then controlBarHeight else 0), (windowWidth - minCodeWidth - minWorkspaceWidth - minToolboxWidth) / canvasAspectRatio)
-    canvasWidth = switch
-      when @level?.isType('web-dev') then windowWidth - minCodeWidth
-      else canvasHeight * canvasAspectRatio
+    if me.get('aceConfig').preferWideEditor or features?.china
```
Comment (mrfinch) [inline app/views/play/level/PlayLevelView.coffee:838]: won't it be always true for china even after setting is changed?

File: app/locale/en.js [hunk 2]
```diff
@@ -1584,6 +1584,8 @@ module.exports = {
       editor_config_indentguides_description: 'Displays vertical lines to see indentation better.',
       editor_config_behaviors_label: 'Enable Smart Behaviors',
       editor_config_behaviors_description: 'Autocompletes brackets, braces, and quotes.',
+      editor_config_wide_editor_label: 'Enable Wide Editor',
+      editor_config_wide_editor_description: 'Make the min size of editor wider like old style',
```
Comment (mrfinch) [inline app/locale/en.js:1588]: Increase the size of code editor

File: app/templates/play/menu/options-view.pug [hunk 3]
```diff
@@ -40,6 +40,14 @@
       span(data-i18n="options.editor_config_behaviors_label") Enable Smart Behaviors
     span.help-block(data-i18n="options.editor_config_behaviors_description") Autocompletes brackets, braces, and quotes.
 
+  .form-group.checkbox
+    label(for="option-wide-editor")
+      input#option-wide-editor(name="wide-editor", type="checkbox", checked=aceConfig.preferWideEditor)
+      span.custom-checkbox
+        .glyphicon.glyphicon-ok
+      span(data-i18n="options.editor_config_wide_editor_label") Enable Wide Editor
```
Comment (mrfinch) [inline app/templates/play/menu/options-view.pug:48]: why have we added strings even though we have their i18n versions?

File: app/views/play/level/PlayLevelView.coffee (hunk 1 above)
Comment (smallst) [reply app/views/play/level/PlayLevelView.coffee:838]: yup it is 😂

File: app/templates/play/menu/options-view.pug (hunk 3 above)
Comment (smallst) [reply app/templates/play/menu/options-view.pug:48]: like a backup. but it is not needed ya

File: app/templates/play/menu/options-view.pug [hunk 4]
```diff
@@ -40,6 +40,15 @@
       span(data-i18n="options.editor_config_behaviors_label") Enable Smart Behaviors
     span.help-block(data-i18n="options.editor_config_behaviors_description") Autocompletes brackets, braces, and quotes.
 
+  if !(features.china)
```
Comment (mrfinch) [inline app/templates/play/menu/options-view.pug:43]: there is such a hatred for the editor size that you don't even want to give option 😂

**Review by mrfinch — APPROVED** [review]

File: app/templates/play/menu/options-view.pug (hunk 4 above)
Comment (smallst) [reply app/templates/play/menu/options-view.pug:43]: haha a unable to get effect option is meaningless and if there's truly someone complain again to ask me give narrow editor back I can update to set default true for china users and enable options. i just don't want to do that for now and get rid of options is easier

**Pull request merged**